android/objectPermission/app/src/main/java/io/realm/todo/PermissionHelper.java

```diff
@@ -12,11 +10,63 @@
 
 public class PermissionHelper {
 
-    // create user specific Role, used to enforce ACL
-    // if it doesn't already exists
-    // should be called from UI thread
-    public static void createUserSpecificRoleIfNotExist(SyncUser user, String roleId, Runnable postStep) {
+    /**
+     * Called after a given user is logged in.
+     * This will ensure the permissions system is setup before starting using the app.
+     * @param user current logged in SyncUser.
+     * @param postInitialization block to run after the Role has been added or found, usually a
+     *                           navigation to the next screen.
+     */
+    public static void initializePermissions(SyncUser user, Runnable postInitialization) {
+        ensureRoleExists(user, postInitialization);// How to close the Realm
```

Comment:
Doesn't this also need to be deferred until after the Realm is downloaded? Otherwise you may end up with the user appearing multiple times on the role. Kinda moot now that the underlying sync issue has been fixed, I guess.